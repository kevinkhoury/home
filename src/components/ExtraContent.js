import React from 'react'
import HTMLContent from './HTMLContent'
import PropTypes from 'prop-types'
import { useInView } from 'react-intersection-observer'

const ExtraContent = ({ content, page, inlineImages }) => {
  const [ref, inView] = useInView({ triggerOnce: true })
  return (
    <section
      ref={ref}
      className={`extra-content extra-content-${page} ${
        inView ? `in-view visible` : 'in-view'
      }`}
    >
      <hr />
      <HTMLContent
        className="extra-content-holder"
        content={content}
        inlineImages={inlineImages}
      />
    </section>
  )
}

ExtraContent.propTypes = {
  content: PropTypes.string.isRequired,
  page: PropTypes.string.isRequired,
  inlineImages: PropTypes.array,
}

export default ExtraContent
