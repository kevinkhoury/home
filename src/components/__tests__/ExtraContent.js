import React from 'react'
import renderer from 'react-test-renderer'
import { render, screen } from '@testing-library/react'
import ExtraContent from '../ExtraContent'
import { mockAllIsIntersecting } from 'react-intersection-observer/test-utils'

jest.mock('../HTMLContent', () => () => 'HTMLContent')

const DEFAULT_PROPS = {
  content: 'foobar',
  page: 'foobar',
  inlineImages: [],
}

afterEach(() => {
  jest.clearAllMocks()
})

describe('ExtraContent', () => {
  it(`renders correctly`, () => {
    const tree = renderer.create(<ExtraContent {...DEFAULT_PROPS} />)
    mockAllIsIntersecting(true)
    expect(tree.toJSON()).toMatchSnapshot()
  })
  it('adds `visible` class when in view', () => {
    render(<ExtraContent {...DEFAULT_PROPS} />)
    mockAllIsIntersecting(true)
    const elem = screen.getByText('HTMLContent')
    expect(elem.classList.contains('visible')).toBe(true)
  })
  it('removes `visible` class when not in view', () => {
    render(<ExtraContent {...DEFAULT_PROPS} />)
    mockAllIsIntersecting(false)
    const elem = screen.getByText('HTMLContent')
    expect(elem.classList.contains('visible')).toBe(false)
  })
})
