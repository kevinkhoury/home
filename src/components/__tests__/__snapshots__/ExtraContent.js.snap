// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExtraContent renders correctly 1`] = `
<section
  className="extra-content extra-content-foobar in-view"
>
  <hr />
  HTMLContent
</section>
`;
